File: core/src/main/java/javabot/web/views/LogsView.java
package javabot.web.views;

import com.google.inject.Injector;
import javabot.dao.LogsDao;
import javabot.model.Logs;
import javabot.web.resources.BotResource;

import javax.inject.Inject;
import javax.servlet.http.HttpServletRequest;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.List;

public class LogsView extends MainView {
    public static final DateTimeFormatter LOG_FORMAT = DateTimeFormatter.ofPattern("hh:mm");
    @Inject
    private LogsDao logsDao;
    private String today;
    private String yesterday;
    private String tomorrow;
    private final String channel;
    private final LocalDateTime date;

    public LogsView(final Injector injector, final HttpServletRequest request, final String channel, final LocalDateTime date) {
        super(injector, request);
        this.channel = channel;
        this.date = date;
        today = BotResource.FORMAT.format(date);
        yesterday = BotResource.FORMAT.format(date.minusDays(1));
        tomorrow = BotResource.FORMAT.format(date.plusDays(1));
    }

    public String getChannel() {
        return channel;
    }

    public String getToday() {
        return today;
    }

    public String getYesterday() {
        return yesterday;
    }

    public String getTomorrow() {
        return tomorrow;
    }

    public String format(LocalDateTime date) {
        return LOG_FORMAT.format(date);
    }

    public List<Logs> logs() {
        return logsDao.findByChannel(channel, date, isAdmin());
    }
    @Override
    public String getChildView() {
        return "logs.ftl";
    }

}

File: src/main/kotlin/javabot/kotlin/web/views/LogsView.kt
package javabot.kotlin.web.views

import com.antwerkz.sofia.Sofia
import com.google.inject.Injector
import javabot.dao.LogsDao
import javabot.dao.util.CleanHtmlConverter
import javabot.kotlin.web.resources.BotResource
import javabot.model.Logs
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import javax.inject.Inject
import javax.servlet.http.HttpServletRequest

public class LogsView(injector: Injector, request: HttpServletRequest, val channel: String, private val date: LocalDateTime) :
      MainView(injector, request) {
    companion object {
        public val LOG_FORMAT: DateTimeFormatter = DateTimeFormatter.ofPattern("hh:mm")
    }

    @Inject
    lateinit var logsDao: LogsDao
    val today = BotResource.FORMAT.format(date)
    val yesterday = BotResource.FORMAT.format(date.minusDays(1))
    val tomorrow = BotResource.FORMAT.format(date.plusDays(1))

    init {
    }

    override fun format(date: LocalDateTime): String {
        return LOG_FORMAT.format(date)
    }

    public fun logs(): List<Logs> {
        val logs = logsDao.findByChannel(channel, date, isAdmin())
        // filter the log content
        for (log in logs) {
            log.message = CleanHtmlConverter.convert(log.message) { s -> Sofia.logsAnchorFormat(s, s) }
        }
        return logs
    }

    override fun getChildView(): String {
        return "logs.ftl"
    }

}
